torchao/quantization/quant_primitives.py

```diff
@@ -256,24 +309,34 @@ def dequantize_affine(
         zero_point = zero_point.view(shape_after_reduction)
 
     if zero_point_domain == ZeroPointDomain.INT:
-        # Force a copy to avoid input modification due
-        # to upcoming in-place operations.
-        dequant = input.to(torch.int32, copy=True)
-        if zero_point is not None:
-            dequant -= zero_point.to(torch.int32)
-        dequant = dequant.to(output_dtype)
-        dequant *= scale
+        if cast_dtypes:
```

Comment:
this is a bit complicated, I'm wondering if we could just do the cast outside of the function as well and remove the `cast_dtype` arg?

Reply:
Any suggestions on how to do this while sharing code? I feel this part is in the middle of the function so it makes it hard to refactor

Reply:
I feel we will need to refactor the code to do the type cast either before or after the entire function